From 100e3aaa92a52e69521ab5d8b9e1f3c376e3097e Mon Sep 17 00:00:00 2001
From: Robert Kanter <rkanter@apache.org>
Date: Tue, 15 Mar 2016 10:05:10 -0700
Subject: [PATCH 1320/2848] YARN-4717.
 TestResourceLocalizationService.testPublicResourceInitializesLocalDir
 fails Intermittently due to
 IllegalArgumentException from cleanup (templedf
 via rkanter)

(cherry picked from commit 22ca176dfe125a4f7bf38cc63ab8106c40a7a7ba)

Change-Id: I44afdc15b45893c2a35dd966f3945dee2e783f2d
---
 .../localizer/TestResourceLocalizationService.java |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/localizer/TestResourceLocalizationService.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/localizer/TestResourceLocalizationService.java
index 1cc5372..e90fc83 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/localizer/TestResourceLocalizationService.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/localizer/TestResourceLocalizationService.java
@@ -193,7 +193,7 @@ public void cleanup() throws IOException {
     conf = null;
     try {
       FileUtils.deleteDirectory(new File(basedir.toString()));
-    } catch (IOException e) {
+    } catch (IOException | IllegalArgumentException e) {
       // ignore
     }
   }
-- 
1.7.9.5

